### ISSUE TITLE
Improve qurl_from_user_input

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#109]
```
def qurl_from_user_input(urlstr):
    """Get a QUrl based on an user input. Additionally handles IPv6 addresses.

    QUrl.fromUserInput handles something like '::1' as a file URL instead of an
    IPv6, so we first try to handle it as a valid IPv6, and if that fails we
    use QUrl.fromUserInput.

    WORKAROUND - https://bugreports.qt-project.org/browse/QTBUG-41089
    FIXME - Maybe https://codereview.qt-project.org/#/c/93851/ has a better way
            to solve this?
```

**Comment by user202729** [comment]
Since the linked QTBUG is closed, this could perhaps just use `QUrl.fromUserInput` directly.

**Comment by The-Compiler** [comment]
True - that got fixed in Qt 5.4 (and the earliest we support is 5.7) - so we can drop `qurl_from_user_input` and use `QUrl.fromUserInput` everywhere instead.

**Comment by The-Compiler** [comment]
qurl_from_user_input was dropped in 0cbd36beb35edc595b0414f14947932495d21c21.
